### DISCUSSION TITLE
Help with Dad Joke API GET Request

### DISCUSSION THREAD

**Discussion by cbjones93** [nss-day-cohort-47/C47-Discussions#27] [Help Me] [answered]
I'm having issues with the joke from chapter Setup Application Data Manager showing up on my console log, I'm getting this in my dev tools: 
dad joke Response {type: "cors", url: "https://icanhazdadjoke.com/j/g3LZTCYDYg", redirected: false, status: 200, ok: true, …}
![image](https://user-images.githubusercontent.com/78928591/110171779-d4f63f00-7dc1-11eb-8e04-a3547fc3b1c8.png)

**Comment by bvinson23** [comment]
So with the dadjoke API, you have to use the Headers tab on Postman and then enter an Accept key with a value of application/json. That will allow you to access their API.

**Comment by BryanNilsen** [comment]
Like what @bvinson23 was saying, you'll need to alter your fetch request to something along the lines of:   
 ```
return fetch("addressToJokeApi", {
        method: "GET",
        headers: {
            "Content-Type": "application/json"
        }
    }
```

**Reply by cbjones93** [reply]
Now I'm getting this.. 
Access to fetch at 'https://icanhazdadjoke.com/' from origin 'http://localhost:5000' has been blocked by CORS policy: Request header field content-type is not allowed by Access-Control-Allow-Headers in preflight response.

I've tried using their custom user-agent to try to get unblocked but no luck.

**Comment by cbjones93** [comment, marked as answer]
so I got this to work, just had to change content-type to accept. Thanks for sending me in the right direction Bryan!

```export const getDadJoke = () =>{
    return fetch ("https://icanhazdadjoke.com",{
        method:"GET",
        headers: {
            "Accept": "application/json"
        }
    })

**Reply by BryanNilsen** [reply]
nice work, @cbjones93 !!
@bvinson23 started us all in the right direction!!
